jmdaemon/jmdaemon/onionmc.py

```diff
@@ -513,33 +515,31 @@ def connect(self) -> None:
                                              self.port)
             self.reconnecting_service = ClientService(onionEndpoint, self.factory)
             # if we want to actually do something about an unreachable host,
-            # we have to force t.a.i.ClientService to give up after the timeout:
+            # we have to force t.a.i.ClientService to give up after the timeout
             d = self.reconnecting_service.whenConnected(failAfterFailures=1)
-            d.addErrback(self.respond_to_connection_failure)
+            d.addCallbacks(self.respond_to_connection_success,
+                           self.respond_to_connection_failure)
             self.reconnecting_service.startService()
 
+    def respond_to_connection_success(self, x):

```

Comment:
Nit: Renaming `x` here would be nice. I see it's an instance of `OnionLineProtocol`, maybe `proto` or `protocol` would work better (somewhere else we call it `p` I think, which I also don't think it's great). We could also add type hinting `-> None`.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1436/commits/f19c0f086f4bcbd9585dda93c870c60430005549